nifi-api/src/main/java/org/apache/nifi/processor/ProcessSession.java

```diff
@@ -109,6 +109,19 @@
     void rollback(boolean penalize);
 
     /**
+     * Produces a listing of pertinent information about the session's
+     * unacknowledged flowfiles that can be used for logging purposes.  If the
+     * session fails for any reason this can help identify problem files with
+     * minimal system impact.
+     *
+     * Because of testing dependencies and ot
+     *
+     * @return {@link String} listing pertinent information about the session's
+     * unacknowledged flowfiles, primarily intended for logging purposes.
+     */
+    String getUnacknowledgedFlowfileInfo();
```

Comment:
When a session gets rolled back everything not committed (even things already transferred but not committed) are part of the list.  Also once a session is rolled back or committed it should generally be able to be used more.  Therefore the information about the session you want to capture needs to come back as a return of the rollback call itself most likely.